apps/studio/src/features/editing-experience/components/form-builder/renderers/controls/JsonFormsNavbarControl/StackableNavbarItem.tsx

```diff
@@ -0,0 +1,342 @@
+  useDisclosure,
+  VStack,
+} from "@chakra-ui/react"
+import { IconButton, Menu } from "@opengovsg/design-system-react"
+import {
+  BiChevronDown,
+  BiDotsHorizontalRounded,
+  BiGridVertical,
+  BiPencil,
+  BiTrash,
+} from "react-icons/bi"
+
+import { DeleteGroupModal } from "./DeleteGroupModal"
+import { DeleteSubItemModal } from "./DeleteSubItemModal"
+
+interface StackableNavbarItemProps {
+  name: string
+  onEdit: (subItemIndex?: number) => void
+  removeItem: (subItemIndex?: number) => void
+  description?: string
+  subItems?: Pick<StackableNavbarItemProps, "name" | "description">[]
+}
+
+export const StackableNavbarItem = ({
+  name,
+  onEdit,
+  removeItem,
+  description,
+  subItems,
+}: StackableNavbarItemProps) => {
+  const {
+    isOpen: isDeleteGroupModalOpen,
+    onOpen: onDeleteGroupModalOpen,
+    onClose: onDeleteGroupModalClose,
+  } = useDisclosure()
+  const {
+    isOpen: isDeleteSubItemModalOpen,
+    onOpen: onDeleteSubItemModalOpen,
+    onClose: onDeleteSubItemModalClose,
+  } = useDisclosure()
+  const [subItemToDelete, setSubItemToDelete] = useState<number>()
+
+  return (
+    <>
+      <DeleteGroupModal
+        label={name}
+        subItemsCount={subItems ? subItems.length : 0}
+        isOpen={isDeleteGroupModalOpen}
+        onClose={onDeleteGroupModalClose}
+        onDelete={() => {
+          removeItem()
+          onDeleteGroupModalClose()
+        }}
+      />
+
+      {!!subItems && (
+        <DeleteSubItemModal
+          label={subItems[subItemToDelete ?? 0]?.name ?? ""}
+          isOpen={isDeleteSubItemModalOpen}
+          onClose={onDeleteSubItemModalClose}
+          onDelete={() => {
+            if (subItemToDelete !== undefined) {
+              removeItem(subItemToDelete)
+              setSubItemToDelete(undefined)
+            }
+
+            onDeleteSubItemModalClose()
+          }}
+        />
+      )}
+
+      <AccordionItem borderTopWidth={0} _last={{ borderBottomWidth: 0 }}>
+        <Box>
+          <Box
+            border="1px solid"
+            borderColor="base.divider.medium"
+            borderRadius="0.375rem"
+            bgColor="utility.ui"
+            transitionProperty="common"
+            transitionDuration="normal"
+            _hover={{
+              bg: "interaction.muted.main.hover",
+              borderColor: "interaction.main-subtle.hover",
+            }}
+            _active={{
+              bg: "interaction.main-subtle.default",
+              borderColor: "interaction.main-subtle.hover",
+              shadow: "0px 1px 6px 0px #1361F026",
+            }}
+          >
+            <HStack gap="0.5rem" p="0.5rem" w="full">
+              <HStack gap="0.75rem" w="full">
+                <Icon as={BiGridVertical} fontSize="1.5rem" color="slate.300" />
+
+                <HStack
+                  as="button"
+                  gap="0.5rem"
+                  w="full"
+                  textAlign="start"
+                  onClick={() => onEdit()}
```

Comment:
nit: `onClick={onEdit}`?

Reply:
Ah cannot, cos `Type '(subItemIndex?: number | undefined) => void' is not assignable to type 'MouseEventHandler<HTMLDivElement>'.`